menpo/io/input/video.py

```diff
@@ -59,53 +28,264 @@ def imageio_ffmpeg_importer(filepath, asset=None, normalise=True, **kwargs):
+    n_frames = round(duration*fps)
+    width = int(video_info['width'])
+    height = int(video_info['height'])
+
+    # Create the resulting dictionary
+    infos = {'duration': duration, 'width': width, 'height': height, 'n_frames': n_frames, 'fps': fps}
+
+    return infos
+
+
+def video_infos_ffprobe(video_filename):
+    """ Parses the information from a video using ffprobe
+    Uses subprocesses to get the information through a pipe
+
+    Parameters
+    ----------
+    video_filename: str
+        absolute path to the video file which information to extract
+
+    Returns
+    -------
+    infos: dict
+        keys are width, height (size of the frames)
+        duration (duration of the video in seconds)
+        n_frames
+    """
+    try:
+        # Seems cleaner to me but adds an unnecessary json dependency
+        import json
+        p = sp.Popen(
+                ['ffprobe', '-show_format', '-show_streams', '-print_format', 'json', video_filename],
+                stdin=sp.PIPE,
+                stdout=sp.PIPE,
+                stderr=sp.PIPE,
+            )
+        # We are reading the data (printed in json format) from the pipe,
+        data = json.loads(''.join(l.decode() for l in p.stdout.readlines()))
+        # For some reason the information is in a list?
+        data['streams'] = data['streams'][0]
+
+    except ImportError:
+        p = sp.Popen(
+            ['ffprobe', '-show_format', '-show_streams', video_filename],
+            stdin=sp.PIPE,
+            stdout=sp.PIPE,
+            stderr=sp.PIPE,
+        )
+        # Store all the information in a dictionary
+        data = dict(streams=[])
+        stream = {}
+        is_stream = is_format = False
+        for line in p.stdout.readlines():
+            # result.append(line)
+            line = line.decode().strip()
+            if line == '[STREAM]':
+                is_stream = True
+                continue
+            if line == '[/STREAM]':
+                is_stream = False
+                data['streams'] = stream
+                stream = {}
+                continue
+            if line == '[FORMAT]':
+                is_format = True
+                continue
+            if line == '[/FORMAT]':
                 break
-            except:
-                pass
-        # Only take the frames before the broken ones
-        ll = ll[:end+1]
+            tokens = line.split('=')
+            if is_stream:
+                stream[tokens[0]] = tokens[1]
+            if is_format:
+                data[tokens[0]] = tokens[1]
 
-    return ll
+    # Keep only the relevant parts
+    width = int(data['streams']['width'])
+    height = int(data['streams']['height'])
+    n_frames = int(data['streams']['nb_frames'])
+    duration = float(data['streams']['duration'])
+    fps = float(eval(data['streams']['avg_frame_rate']))
+
+    infos = {'duration': duration, 'width': width, 'height': height, 'n_frames': n_frames, 'fps': fps}
+
+    return infos
```

Comment:
@JeanKossaifi super minor, but end files with a empty blank line:
```
}

```
not
```
}
```
more info: http://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file